handler.go

```diff
@@ -0,0 +1,269 @@
+		if err = rows.StructScan(&login); err != nil {
+			return nil, err
+		}
+
+		logins = append(logins, login)
+	}
+
+	return logins, nil
+}
+
+func loginCount(dbconn *sqlx.DB, userID string) (uint, error) {
+	var err error
+
+	usersT := goqu.T("users")
+	loginsT := goqu.T("logins")
+	countQ := goqu.From(loginsT).
+		Join(usersT, goqu.On(loginsT.Col("user_id").Eq(usersT.Col("id")))).
+		Where(usersT.Col("id").Eq(userID)).
+		Select(goqu.COUNT("*"))
+	q, _, err := countQ.ToSQL()
+	if err != nil {
+		return 0, err
+	}
+
+	var count uint
+	if err = dbconn.QueryRowxContext(context.Background(), q).Scan(&count); err != nil {
+		return count, err
+	}
+	return count, nil
+}
+
+func getHandler(conn *nats.EncodedConn, dbconn *sqlx.DB) nats.Handler {
+	return func(subject, reply string, request *user.UserLookupRequest) {
+		var err error
+
+		carrier := gotelnats.PBTextMapCarrier{
+			Header: request.Header,
+		}
+
+		ctx, span := gotelnats.StartSpan(&carrier, subject, gotelnats.Process)
+		defer span.End()
+
+		usersT := goqu.T("users")
+		jobsT := goqu.T("jobs")
+		savedSearchesT := goqu.T("user_saved_searches")
+		prefsT := goqu.T("user_preferences")
+
+		q := goqu.From(usersT)
+
+		log.Infof("%+v\n", request)
+
+		if request.IncludeSavedSearches {
+			q = q.Join(savedSearchesT, goqu.On(usersT.Col("id").Eq(savedSearchesT.Col("user_id"))))
+		}
+
+		if request.IncludePreferences {
+			q = q.Join(prefsT, goqu.On(usersT.Col("id").Eq(prefsT.Col("user_id"))))
+		}
+
+		switch x := request.LookupIds.(type) {
+		case *user.UserLookupRequest_AnalysisId:
+			analysisID := request.GetAnalysisId()
+
+			q = q.
+				Join(jobsT, goqu.On(usersT.Col("id").Eq(jobsT.Col("user_id")))).
+				Where(jobsT.Col("id").Eq(analysisID))
+
+		case *user.UserLookupRequest_Username:
+			q = q.Where(usersT.Col("username").Eq(request.GetUsername()))
+
+		case *user.UserLookupRequest_UserId:
+			q = q.Where(usersT.Col("id").Eq(request.GetUserId()))
+
+		default:
+			lookupErr := fmt.Errorf("lookup type %T not known", x)
+			handleError(ctx, lookupErr, svcerror.Code_BAD_REQUEST, reply, conn)
+			return
+		}
+
+		selectFields := []interface{}{
+			usersT.Col("id"),
+			usersT.Col("username"),
+		}
+
+		if request.IncludeSavedSearches {
+			selectFields = append(
+				selectFields,
+				savedSearchesT.Col("id").As("saved_searches_id"),
+				savedSearchesT.Col("saved_searches").As("saved_searches"),
+			)
+		}
+
+		if request.IncludePreferences {
+			selectFields = append(
+				selectFields,
+				prefsT.Col("id").As("preferences_id"),
+				prefsT.Col("preferences").As("preferences"),
+			)
+		}
+
+		q = q.Select(selectFields...)
+
+		queryString, _, err := q.ToSQL()
+		if err != nil {
+			handleError(ctx, err, svcerror.Code_INTERNAL, reply, conn)
+		}
+
+		log.Infof("%s\n", queryString)
+
+		u := lookupUser{}
+
+		// argument handling is done by the goqu library above.
+		if err = dbconn.QueryRowxContext(context.Background(), queryString).StructScan(&u); err != nil {
```

Comment:
Huh, does `goqu` not return the args so you pass them in parameterized like other things? That's a bit unfortunate if so, lot better to trust postgresql with avoiding SQL injection IMO